pkg/apis/v1/labels.go

```diff
@@ -51,6 +51,7 @@ const (
 	NodePoolHashAnnotationKey                  = apis.Group + "/nodepool-hash"
 	NodePoolHashVersionAnnotationKey           = apis.Group + "/nodepool-hash-version"
 	NodeClaimTerminationTimestampAnnotationKey = apis.Group + "/nodeclaim-termination-timestamp"
+	NodeClaimSyncTaintsAnnotationKey           = apis.Group + "/nodeclaim-sync-taints"
```

Comment:
This needs to be label, no? Kubelet configuration does not allow nodes to be registered with annotations

Reply:
This isn't going to be coming from the kubelet. It would come from the provider, I thought. This is an annotation on the NodeClaim, not the Node.

Reply:
So in this case, the provider should be using kubelet configuration is the way that will ensure that taint is always added to the node on registration. We will use `--node-labels` flag for the kubelet configuration, which will tell Karpenter if the startup taint is already added or not https://kubernetes.io/docs/reference/command-line-tools-reference/kubelet/